### DISCUSSION TITLE
Executing cake script from PS bootstrapper prompts full framework 4.6.1 install prompt

### DISCUSSION THREAD

**Discussion by MeikTranel** [cake-build/cake#3529] [Q&A]
### What You Are Seeing?

![image](https://user-images.githubusercontent.com/6466560/34101811-39d88bda-e3e7-11e7-859c-4b10f671e396.png)

### What is Expected?

Cake starts compiling, gathering tools and run the script

### What version of Cake are you using?
0.23

### Are you running on a 32 or 64 bit system?

64-bit

### What environment are you running on?  Windows? Linux? Mac?

Windows 8.1 Pro

![image](https://user-images.githubusercontent.com/6466560/34102354-12471c6a-e3e9-11e7-9e55-d963a7b47883.png)

### How Did You Get This To Happen? (Steps to Reproduce)

```
var solutions = new string[] {
 "../solutionA.sln",
 "../solutionB.sln"
};

Task("Default")
.DoesForEach(
    solutions,
    solution => {
        NuGetRestore(solution,new NuGetRestoreSettings{
            PackagesDirectory = "../Packages/",
            ToolPath = "../.nuget/NuGet.exe"
        });
        MSBuild(solution, new MSBuildSettings() {
            Verbosity = Verbosity.Minimal
        });
    }
);

Task("Test")
.IsDependentOn("Default")
.Does(() => {
    MSTest("PATH/TO/ASSEMBLY.dll");
});

RunTarget("Default");
```

### Output Log
```
PS C:\Users\build\Desktop\test\.build> .\build.ps1
Preparing to run build script...
Running build script...
PS C:\Users\build\Desktop\test\.build>
```

All attempts (vanilla, Verbosity, Verbosity+mono, Verbosity+Experimental) produce the error prompt

**Comment by rroman81** [comment]
any workarounds?  Does installing the .net 4.6.1 developer pack helped?

**Comment by devlead** [comment]
.NET 4.6.1 is the minimal supported .NET Runtime needed for Roslyn now, so that's why it's also needed by Cake, so no workaround avail, installing .NET runtime 4.6.1 or newer will solve it.

**Comment by rroman81** [comment]
well, i have a colleague that has .net 4.6.2 installed and getting the same prompt.  I am trying to figure out what's missing for 4.6.1 that it's asking for 4.6.1
